enacts/wat_bal/maproom_monit.py

```diff
@@ -549,7 +556,10 @@ def wat_bal_tile(tz, tx, ty):
     p_d = calc.sel_day_and_month(
         precip["T"], int(planting_day), planting_month1
     ).squeeze(drop=True).rename("p_d")
-    precip = precip.sel(T=slice(p_d.dt.strftime("%-d %b %y"), the_date))
+    precip = precip.sel(T=slice(
+        (p_d - np.timedelta64(7 - 1, "D")).dt.strftime("%-d %b %y"),
+        the_date,
+    ))
```

Comment:
You had to make this change in two places. Consider factoring out the calculation that is common to wat_bal_ts and wat_bal_tile.

Reply:
yes... but there isn't much more than that in common. wat_bal_ts takes a slice from p_d - 6 to today; whereas wat_bal_tile takes a slice from p_d - 6 to user-chosen date (located somewhere between p_d and today). And the logic to identify p_d, and slice things are different in each

Reply:
It's more complicated than I realized because there are three different variants of the calculation: for the map, for the monitoring curve, and for the reference year curve. Nonetheless, Rémi will look into factoring out common code.

Reply:
I factored out up to a certain point. Passed that point, peffective and application of the spwb function are probably factorable out, but there are things in the way that I am not sure what they are for, and/or where they can/must be. These are a load and a try in the ts section, and there is all the tiling business in the map section.

Reply:
I would like to further reduce the duplication between `wat_bal_tile` and `wat_bal_ts`.

We can do the `load` in both cases (or in neither if you find that it makes no difference to performance).

The try/except in `wat_bal_ts` seems to have been written with a specific case in mind, and I can't guess what that is (a comment would have been useful). Whether it belongs inside or outside the factored-out function depends on whether the error happens in the calculation part or not. I suggest just getting rid of the try/except, doing the refactoring, and then putting back the try/except in the right place when we see the error again.

Another difference is that `wat_bal_ts` gets `taw` as a parameter, while `wat_bal_tile` reads it from a file. That seems like an arbitrary and unnecessary difference.

Reply:
Done. There is probably a need to fix the data reading as you did for flex. fcst. but I'd rather have that be a separate PR.

_ts and _tile differ a bit because _tile is what is directly called by the Callbacks whereas _ts is called (twice) by _plots that is the Callback one and where  raw is read from file.